pyquil/paulis.py

```diff
@@ -289,11 +316,35 @@ def __init__(self, terms):
         """
         :param list terms: A list of PauliTerms.
         """
+        if not (isinstance(terms, list) and all([isinstance(term, PauliTerm) for term in terms])):
```

Comment:
list is fairly restrictive. Try `collections.Sequence` which should cover lists, tuples and numpy arrays

Reply:
I was just checking for what the docstring said, I wasn't sure if it was assumed to be a list somewhere else.

Reply:
Of course, given that I decided to touch this code, I suppose following through is the least I can do...